src/config/scrivitoSites.ts

```diff
@@ -20,19 +26,27 @@ export function baseUrlForSite(siteId: string): string | undefined {
 export function siteForUrl(
   url: string,
 ): { baseUrl: string; siteId: string } | undefined {
+  const neoletterMailingsBaseUrl = baseUrlForSite(NEOLETTER_MAILINGS_SITE_ID)
+  if (neoletterMailingsBaseUrl && url.startsWith(neoletterMailingsBaseUrl)) {
+    return {
+      baseUrl: neoletterMailingsBaseUrl,
+      siteId: NEOLETTER_MAILINGS_SITE_ID,
+    }
+  }
+
   const language = /\b\/([0-9a-f]{32}\/)?(?<lang>[a-z]{2})([?/]|$)/.exec(url)
     ?.groups?.lang
 
-  const siteId = Obj.onAllSites()
+  return Obj.onAllSites()
     .where('_path', 'equals', '/')
     .and('_language', 'equals', language || null)
-    .first()
-    ?.siteId()
-
-  if (!siteId) return
-
-  const baseUrl = baseUrlForSite(siteId)
-  if (baseUrl) return { baseUrl, siteId }
+    .toArray()
+    .map((root) => {
+      const siteId = root.siteId()
+      const baseUrl = siteId && baseUrlForSite(siteId)
+      if (baseUrl) return { baseUrl, siteId }
+    })
+    .find((site) => site)
```

Comment:
I'm still having a hard time to understand, what this this is actually about. Why do we need this?

Reply:
We actually don't need the filtering anymore. It was introduced when we ignored the mailings site. We can now revert to an earlier implementation.